go/vt/mysqlctl/replication.go

```diff
@@ -377,7 +377,7 @@ func (mysqld *Mysqld) WaitSourcePos(ctx context.Context, targetPos replication.P
 	if result.IsNull() {
 		return fmt.Errorf("%v(%v) failed: replication is probably stopped", waitCommandName, query)
 	}
-	if result.ToString() == "-1" {
+	if result.ToString() == "1" {
```

Comment:
I think we should invert the check here and validate success. 

For any functions that return C style integer arguments where one value specifically indicates success, we should always check against that value. I’ve see trying to check against failure values to cause problems too many times.  

So either here do a != “0” or switch the return and fallback and do == “0”.

Reply:
Yeah, this way would work with 8.0 and 8.2, right? 0 is success, non-zero value is failure. And I think that we should add a const for it so that it's more obvious.

Reply:
Good points. I'll push a change.

Reply:
See later comments from Rohit.